src/Event/Http/FpmHandler.php

```diff
@@ -182,6 +182,11 @@ private function eventToFastCgiRequest(HttpRequestEvent $event, Context $context
         $request->setCustomVar('LAMBDA_INVOCATION_CONTEXT', json_encode($context));
         $request->setCustomVar('LAMBDA_REQUEST_CONTEXT', json_encode($event->getRequestContext()));
 
+        if (isset($event->toArray()['identity']['sourceIp'])) {
+            // Add user's IP as an easy to access variable
+            $request->setCustomVar('LAMBDA_IDENTITY_SOURCE_IP', $event->toArray()['identity']['sourceIp']);
+        }
```

Comment:
Shouldn't we put that IP address directly into `$request->setRemoteAddress(...);` above?

Reply:
No, I dont think we should. 

RemoteAddress is the last proxy. We are behind at least 2 proxies and we should not confuse things more that needed. 

Yes, it is super confusing to properly get the user IP in PHP. Here is how I will write in my index.php after this PR is merged. 


```
<?php

use App\Kernel;
use Symfony\Component\ErrorHandler\Debug;
use Symfony\Component\HttpFoundation\Request;

require dirname(__DIR__).'/config/bootstrap.php';

// Mark 127.0.0.1 as a trusted proxy
Request::setTrustedProxies(['127.0.0.1']), Request::HEADER_X_FORWARDED_ALL ^ Request::HEADER_X_FORWARDED_HOST);

// It is only trusted on this host
Request::setTrustedHosts(['api.tnyholm.se']);

// Get user IP:
$_SERVER['HTTP_X_FORWARDED_FOR'] = $_SERVER['LAMBDA_IDENTITY_SOURCE_IP'] ?? '';

$kernel = new Kernel($_SERVER['APP_ENV'], (bool) $_SERVER['APP_DEBUG']);
$request = Request::createFromGlobals();
$response = $kernel->handle($request);
$response->send();
$kernel->terminate($request, $response);
```

This will tell Symfony to trust localhost and the `HTTP_X_FORWARDED_FOR` header. 
I can now run `$request->getClientIp()` to get the correct IP. 

Since proxies are complicated and Im no expert, I did not want to make a change to Bref to modify `HTTP_X_FORWARDED_FOR` or the `setRemoteAddress`. 

----

Note also that if we were changing `$request->setRemoteAddress(...)`, that would break BC.

Reply:
Right, that makes sense thanks.

Sorry for delaying this, but another question then: would it make more sense to get the client's IP address from the X-Forwarded-For headers instead?

I'm just making sure that there isn't a more standard solution for this. Ideally getting the IP address of the client shouldn't be specific to Bref or AWS Lambda.

Reply:
> would it make more sense to get the client's IP address from the X-Forwarded-For headers instead?

I think so. But Im not sure. That is why I went with the more safe option for Bref in case Im wrong.

Reply:
Right. I am cautious about adding this because:

- removing it in the future is a BC break
- it is Bref-specific

That's why I'm wondering about those headers. After all, the same problem would exist with any PHP application behind a proxy. And since you already call `Request::setTrustedProxies()`, it might already work.